src/net_processing.cpp

```diff
@@ -1214,6 +1226,40 @@ void PeerLogicValidation::NewPoWValidBlock(const CBlockIndex *pindex, const std:
     });
 }
 
+static bool UpdateCFHeadersCache(const BlockFilterIndex& filter_index)
+{
+    LOCK(cs_main);
+    std::lock_guard<std::mutex> _lock(active_chain_cf_headers_mtx);
```

Comment:
Kinda off-topic I guess, bug locking two mutex'es makes you think, maybe bitcoin codebase would need something like `std::lock`, or better `std::scoped_lock` (C++17) to lock multiple locks in deadlock-free way, with same LOCK() debugging features.

Reply:
This doesn't seem actionable here. Probably best to open an issue or discuss in IRC or something?